backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTypesCreator.java

```diff
@@ -39,9 +42,10 @@ public CassandraTypesCreator(CassandraModule module, CqlSession session) {
     public InitializationStatus initializeTypes() {
         KeyspaceMetadata keyspaceMetadata = session.getMetadata().getKeyspaces().get(session.getKeyspace().get());
 
-        return types.stream()
-                .map(type -> type.initialize(keyspaceMetadata, session))
-                .reduce(InitializationStatus::reduce)
-                .orElse(InitializationStatus.ALREADY_DONE);
+        return Flux.fromIterable(types)
+            .flatMap(type -> type.initialize(keyspaceMetadata, session))
+            .reduce(InitializationStatus::reduce)
```

Comment:
nitpick: I would rather use the reduce with initial state for clarity

Reply:
The semantic is actually different and is not straightforward with the syntax you propose. I will keep this as is.